components/event-processor/org.wso2.carbon.event.processor.core/src/main/java/org/wso2/carbon/event/processor/core/internal/util/helper/EventProcessorHelper.java

```diff
@@ -208,14 +208,29 @@ private static boolean validateSiddhiStreamWithDatabridgeStream(String streamNam
             if (streamDefinition != null) {
                 String siddhiAttributeName;
                 int attributeCount = 0;
+                int siddhiStreamAttributeCount = siddhiStreamDefinition.getAttributeList().size();
                 int streamSize = (streamDefinition.getMetaData() == null ? 0 : streamDefinition.getMetaData().size())
                         + (streamDefinition.getCorrelationData() == null ? 0 : streamDefinition.getCorrelationData().size())
                         + (streamDefinition.getPayloadData() == null ? 0 : streamDefinition.getPayloadData().size());
-                if (siddhiStreamDefinition.getAttributeList().size() != streamSize) {
+                Element ArbitraryElement = AnnotationHelper.getAnnotationElement(EventProcessorConstants.ANNOTATION_IMPORT,
+                        EventProcessorConstants.ANNOTATION_INCLUDE_ARBITRARY, siddhiStreamDefinition.getAnnotations());
+                if (ArbitraryElement == null) {
+                    ArbitraryElement = AnnotationHelper.getAnnotationElement(EventProcessorConstants.ANNOTATION_EXPORT,
+                            EventProcessorConstants.ANNOTATION_INCLUDE_ARBITRARY, siddhiStreamDefinition.getAnnotations());
+                }
+                if (ArbitraryElement != null && ArbitraryElement.getValue().equals("true")) {
+                    if (!siddhiStreamDefinition.getAttributeList().get(siddhiStreamAttributeCount - 1).getName().equals(EventProcessorConstants.ARBITRARY_MAP)) {
```

Comment:
Hi Tishan,

This is fixed by the commits in : https://github.com/wso2/carbon-event-processing/pull/266

Thanks,
SajithD